libs/cardano-ledger-core/src/Cardano/Ledger/BaseTypes.hs

```diff
@@ -176,6 +177,9 @@ import NoThunks.Class (NoThunks (..))
 import Numeric.Natural (Natural)
 import Quiet (Quiet (Quiet))
 
+maxScale :: Int
+maxScale = 19
```

Comment:
We should not export this biding from the module and it should have a better name, along the line of:
```suggestion
numFullDigitsWord64 :: Int
numFullDigitsWord64 = 19
```

Reply:
Renamed it. The "full digits" part sounds strange to me but including `Word64` in the name is good. `maxScale` could indeed be a bit misleading without context.

Reply:
`maxBound` for `Word64` has 20 digits:
```haskell
ghci> length (show (maxBound :: Word))
20
```

But with regards to representing decimal numbers it has only 19 digits available:
```
ghci> (2 * 10 ^ 19) > toInteger (maxBound :: Word)
True
ghci> (1 * 10 ^ 19) > toInteger (maxBound :: Word)
False
```
For example, the smallest decimal number you can represent with `Ratio Word64` is this number that has 19 decimal places (despite that numerator has 20 digits :grin:):
```
ghci> 0.0000000000000000001 :: Ratio Word
1 % 10000000000000000000
ghci> 0.00000000000000000001 :: Ratio Word
1 % 7766279631452241920
```

With all this information at your disposal, if you have a better name for the binding, please go for it.

Reply:
Yes, this is what I added to the commit body here: be55cbc14d2fc61970b711d77eb857b70e8c896b . Although not in such detail. :D This is why I picked `maxScale` since you refer to that parameter as scale too in `safe-decimal`: https://hackage.haskell.org/package/safe-decimal-0.2.1.0/docs/Numeric-Decimal.html

How about `maxScaleWord64`? Or `maxDecimalsWord64` since mathematics sometimes refer to the decimal places after the decimal point as decimals? I think this latter one is more well-known.

Reply:
`maxDecimalsWord64` sounds ideal!